api/axis.py

```diff
@@ -448,6 +461,11 @@ def fulfill(self, patron, pin, licensepool, internal_format, **kwargs):
             fulfillment = loan.fulfillment_info
             if not fulfillment or not isinstance(fulfillment, FulfillmentInfo):
                 raise CannotFulfill()
+
+            if isinstance(patron, Patron) and loan.external_identifier:
+                # Save the external_identifier to the Patron's Loan for future retrieval
+                licensepool.loan_to(patron, external_identifier=loan.external_identifier)
```

Comment:
This usage of `loan_to` isn't as bad because if you were able to fulfill a loan, you really should have a local Loan object by now. But I do think it would be better to fetch the loan (using get_one like in my earlier suggestion) and skip this step if for whatever reason (maybe a race condition) it's not found in the database. If nothing else, calling `loan_to` overwrites `Loan.start` and makes it look like the loan started right now.